### PR TITLE
Add support for HTMLEditor when webengine is not available for Qt

### PR DESCRIPTION
**Pull request by corranwebster**
This is a completely untested attempt to add basic support for the HTMLeditor when the advanced web widgets are not available.

This is a potential start on a fix for #1463

### CODE REVIEW

**Comment by corranwebster** [comment]
Testing this on PySide2 without WebEngine, tests fail because the QTextBrowser API is different from the QWebView API, so full PR will require changes to the testing infrastructure for the html editor as well.

Manual testing shows that it at least works as advertised.

File: traitsui/qt4/html_editor.py [hunk 1]
```diff
@@ -95,10 +109,15 @@ def dispose(self):
...
-                # QtWebKit-only cleanup
-                page.linkClicked.disconnect(self._link_clicked)
+            if HAS_WEB_VIEW:
+                page = self.control.page()
+                if hasattr(page, 'setLinkDelegationPolicy'):
+                    # QtWebKit-only cleanup
+                    page.linkClicked.disconnect(self._link_clicked)
+            else:
+                # QTextBrowser clean-up
+                self.control.anchorClicked.connect(self._link_clicked)
```
Comment (rahulporuri) [inline traitsui/qt4/html_editor.py:119]:
```suggestion
                self.control.anchorClicked.disconnect(self._link_clicked)
```

**Review by rahulporuri — APPROVED** [review]
LGTM. I tested this locally by installing `pyside2` from pypi and launching the etsdemo application. I was able to open the standard editor demos - and click on links. Clicking on links opened the doc page in my default browser, which IIUC is the desired and implemented behavior here.

I can't think of alternative solutions so this is the best one we have at the moment.

**Comment by aaronayres35** [comment]
I can confirm that running the test suite with `pyside2` from edm I see:
<details>

```
======================================================================
ERROR: test_open_external_link (traitsui.tests.editors.test_html_editor.TestHTMLEditor)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/aayres/Desktop/traitsui/traitsui/tests/editors/test_html_editor.py", line 307, in test_open_external_link
    html_view.perform(MouseClick())
  File "/Users/aayres/Desktop/traitsui/traitsui/testing/tester/ui_wrapper.py", line 248, in perform
    self._perform_or_inspect(interaction)
  File "/Users/aayres/Desktop/traitsui/traitsui/testing/tester/ui_wrapper.py", line 313, in _perform_or_inspect
    return handler(self, interaction)
  File "/Users/aayres/Desktop/traitsui/traitsui/tests/editors/test_html_editor.py", line 202, in <lambda>
    wrapper._target.control, wrapper.delay
  File "/Users/aayres/Desktop/traitsui/traitsui/tests/editors/test_html_editor.py", line 168, in qt_mouse_click_web_view
    qt_allow_view_to_load(view)
  File "/Users/aayres/Desktop/traitsui/traitsui/tests/editors/test_html_editor.py", line 150, in qt_allow_view_to_load
    wait_for_qt_signal(loadable.loadFinished, timeout=timeout_ms)
AttributeError: 'PySide2.QtWidgets.QTextBrowser' object has no attribute 'loadFinished'

======================================================================
ERROR: test_open_external_link_externally (traitsui.tests.editors.test_html_editor.TestHTMLEditor)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/aayres/Desktop/traitsui/traitsui/tests/editors/test_html_editor.py", line 368, in test_open_external_link_externally
    html_view.perform(MouseClick())
  File "/Users/aayres/Desktop/traitsui/traitsui/testing/tester/ui_wrapper.py", line 248, in perform
    self._perform_or_inspect(interaction)
  File "/Users/aayres/Desktop/traitsui/traitsui/testing/tester/ui_wrapper.py", line 313, in _perform_or_inspect
    return handler(self, interaction)
  File "/Users/aayres/Desktop/traitsui/traitsui/tests/editors/test_html_editor.py", line 202, in <lambda>
    wrapper._target.control, wrapper.delay
  File "/Users/aayres/Desktop/traitsui/traitsui/tests/editors/test_html_editor.py", line 168, in qt_mouse_click_web_view
    qt_allow_view_to_load(view)
  File "/Users/aayres/Desktop/traitsui/traitsui/tests/editors/test_html_editor.py", line 150, in qt_allow_view_to_load
    wait_for_qt_signal(loadable.loadFinished, timeout=timeout_ms)
AttributeError: 'PySide2.QtWidgets.QTextBrowser' object has no attribute 'loadFinished'

======================================================================
ERROR: test_open_internal_link (traitsui.tests.editors.test_html_editor.TestHTMLEditor)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/aayres/Desktop/traitsui/traitsui/tests/editors/test_html_editor.py", line 281, in test_open_internal_link
    html_view.perform(MouseClick())
  File "/Users/aayres/Desktop/traitsui/traitsui/testing/tester/ui_wrapper.py", line 248, in perform
    self._perform_or_inspect(interaction)
  File "/Users/aayres/Desktop/traitsui/traitsui/testing/tester/ui_wrapper.py", line 313, in _perform_or_inspect
    return handler(self, interaction)
  File "/Users/aayres/Desktop/traitsui/traitsui/tests/editors/test_html_editor.py", line 202, in <lambda>
    wrapper._target.control, wrapper.delay
  File "/Users/aayres/Desktop/traitsui/traitsui/tests/editors/test_html_editor.py", line 168, in qt_mouse_click_web_view
    qt_allow_view_to_load(view)
  File "/Users/aayres/Desktop/traitsui/traitsui/tests/editors/test_html_editor.py", line 150, in qt_allow_view_to_load
    wait_for_qt_signal(loadable.loadFinished, timeout=timeout_ms)
AttributeError: 'PySide2.QtWidgets.QTextBrowser' object has no attribute 'loadFinished'

======================================================================
ERROR: test_open_internal_link_externally (traitsui.tests.editors.test_html_editor.TestHTMLEditor)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/aayres/Desktop/traitsui/traitsui/tests/editors/test_html_editor.py", line 341, in test_open_internal_link_externally
    html_view.perform(MouseClick())
  File "/Users/aayres/Desktop/traitsui/traitsui/testing/tester/ui_wrapper.py", line 248, in perform
    self._perform_or_inspect(interaction)
  File "/Users/aayres/Desktop/traitsui/traitsui/testing/tester/ui_wrapper.py", line 313, in _perform_or_inspect
    return handler(self, interaction)
  File "/Users/aayres/Desktop/traitsui/traitsui/tests/editors/test_html_editor.py", line 202, in <lambda>
    wrapper._target.control, wrapper.delay
  File "/Users/aayres/Desktop/traitsui/traitsui/tests/editors/test_html_editor.py", line 168, in qt_mouse_click_web_view
    qt_allow_view_to_load(view)
  File "/Users/aayres/Desktop/traitsui/traitsui/tests/editors/test_html_editor.py", line 150, in qt_allow_view_to_load
    wait_for_qt_signal(loadable.loadFinished, timeout=timeout_ms)
AttributeError: 'PySide2.QtWidgets.QTextBrowser' object has no attribute 'loadFinished'
```

</details>

whereas with `pyside2` from PyPI the tests pass.

**Comment by aaronayres35** [comment]
Also, I tried testing etsdemo with `pyside2` from PyPI and as mentioned here: https://github.com/enthought/traitsui/pull/1504#pullrequestreview-647882443 things work as expected.

When testing with pyside2 from edm I was having problems.
I ran:
```
edm envs create dumb --force
edm shell -e dumb
pip install -e . # from traitsui root
edm install pyside2 pygments
cd ets-demo
pip install -e .
cd ..
etsdemo
```
and when clicking on a StandardEditors example I see (with a print statement of `self.control` inside `update_editor`):
```
$ etsdemo 
2021-05-03 11:04:32.106 Python[12666:3110140] ApplePersistenceIgnoreState: Existing state will not be touched. New state will be written to (null)
<PySide2.QtWidgets.QTextBrowser(0x7fae2170e490) at 0x7fae2801ea88>
Exception occurred in traits notification handler for object: <traitsui.qt4.html_editor.SimpleEditor object at 0x7fae2675c308>, trait: base_url, old value: , new value: /Users/aayres/.edm/envs/dumby/lib/python3.6/site-packages/traits/examples/introduction
Traceback (most recent call last):
  File "/Users/aayres/.edm/envs/dumby/lib/python3.6/site-packages/traits/trait_notifiers.py", line 342, in __call__
    self.handler(*args)
  File "/Users/aayres/Desktop/traitsui/traitsui/qt4/html_editor.py", line 142, in _base_url_changed
    self.update_editor()
  File "/Users/aayres/Desktop/traitsui/traitsui/qt4/html_editor.py", line 135, in update_editor
    self.control.setHtml(text, QtCore.QUrl.fromLocalFile(url))
TypeError: setHtml() takes exactly one argument (2 given)
...
```

**Comment by aaronayres35** [comment]
I've made a fix for https://github.com/enthought/traitsui/pull/1504#issuecomment-831363849, but the test suite still needs to be updated.  We likely need to special handle this case as the other tests might not really apply, or will need to modified to some extent.  Additionally the UITester registry will need to be updated eventually.

For right now I have simply skipped the tests if neither QtWebkit or QtWebEngine is available.  We probably don't want to push this through until the tests get sorted out

**Review by rahulporuri — APPROVED** [review]
LGTM. I tested this against pyside2 from edm and pypi - and it works as expected with both.

**Pull request merged**